### DISCUSSION TITLE
Node Id as an entity

### DISCUSSION THREAD

**Discussion by alexruffell** [zwave-js/zwave-js-ui#4041] [Ideas]
I am not sure what I suggest in the title is the correct solution, but these are my use case scenarios:

1) Zwave Events for scene control using Inovelli switches in Node Red: I believe the only way to capture the scene events from zwave switches is by looking at Zwave Events as shown in the snapshot below. Unfortunately I cannot specify the entity or device as that information is not present in the data. Only the node id is there so I have to filter using that (yellow node). The issue is that every time I make a change that results in the node id changing, I have to adjust the flows. What I am looking for is some way to resolve the node id by device or entity (a DNS of sorts for Zwave devices) and I am guessing that exposing the node id as an entity would help with that. While the issue above is not major, having this ability would make automations that relay on watching for zwave events more resilient...

<img width="836" alt="image" src="https://github.com/user-attachments/assets/e72f1f4f-59c8-491f-968a-99c0584097f5">

The yellow node manually assigns the node id I have to update when I make changes to 'msg.payload.event.node_id'

2) Since Zwave JS hides the Node Id (inside node page, drill down menu), I often refer to ZWave JS UI simply to get the Node Id. One view I created for my Zigbee devices is very useful when I need to troubleshoot things, but unfortunately I cannot replicate it for Zwave. One of the missing elements is the Node Id. This is a snippet of the Zigbee table I'd like to replicate for Zwave:

<img width="1590" alt="image" src="https://github.com/user-attachments/assets/f6b69ae4-b31c-48c9-8f30-5e6fea2e02f8">

**Comment by kpine** [comment]
Are you using MQTT or Z-Wave integration in HA? For the latter, ZUI has no influence on entities created in HA.
